vnext/Shared/HermesSamplingProfiler.cpp

```diff
@@ -0,0 +1,54 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+#include "pch.h"
+
+#include <chrono>
+
+#include <winrt/Windows.Storage.h>
+#include <hermes/hermes.h>
+#include "Unicode.h"
+
+#include "HermesSamplingProfiler.h"
+
+namespace Microsoft::ReactNative {
+
+namespace {
+std::future<std::string> getTraceFilePath() {
+  auto hermesFolder = (co_await winrt::Windows::Storage::ApplicationData::Current().LocalFolder().CreateFolderAsync(
```

Comment:
note that unpackaged win32 apps won't quite know what to do with this so we may need fallbacks to use non WinRT file I/O APIs (see Async Storage module for an example)

Reply:
Alex, I'd like to defer making the sampler win32 friendly in a subsequent PR. Hope it works !